### PR TITLE
ENH: webdiff git driver added

### PR DESCRIPTION
**Pull request by tworec**
This is my first contribution to jupyter family.
I've added webdiff driver for git.

### CODE REVIEW

**Comment by vidartf** [comment]
Thanks for the contribution! In terms of the functionality, this is something that would be welcome. In terms of the implementation I'll do a more detailed review later on, and try to fix that CI failure. Hopefully next week, if not then the week after!

**Comment by tworec** [comment]
cool, thx.
I'm eager to get detailed feedback from you.
be well

File: nbdime/args.py [hunk 1]
```diff
@@ -205,16 +209,18 @@ def add_diff_args(parser):
...
-    diff_parser.add_argument('b_sha1', nargs='?', default=None)
-    diff_parser.add_argument('b_mode', nargs='?', default=None)
-    diff_parser.add_argument('rename_to', nargs='?', default=None)
+    diff_parser.add_argument('base', nargs='?', default=None)
+    diff_parser.add_argument('base_sha1', nargs='?', default=None, action=SkipAction)
+    diff_parser.add_argument('base_mode', nargs='?', default=None, action=SkipAction)
+    diff_parser.add_argument('remote', nargs='?', default=None)
+    diff_parser.add_argument('remote_sha1', nargs='?', default=None, action=SkipAction)
+    diff_parser.add_argument('remote_mode', nargs='?', default=None, action=SkipAction)
+    diff_parser.add_argument('rename_to', nargs='?', default=None, action=SkipAction)
```
Comment (tworec) [inline nbdime/args.py]: cute

File: nbdime/gitdiffdriver.py [hunk 2]
```diff
@@ -96,19 +95,23 @@ def main(args=None):
...
+        return nbdiffapp.main_diff(opts)
     elif opts.subcommand == 'webdiff':
-        from nbdime.webapp import nbdiffweb
-        return nbdiffweb.main(
-            [opts.a, opts.b] +
-            ['--%s' % name for name in diff_exclusives if getattr(opts, name)] +
-            [i for l in [['--%s' % arg, str(getattr(opts, arg.replace('-', '_')))] for arg in web_args] for i in l]
-            )
+        from .webapp import nbdiffweb
+        return nbdiffweb.main_diff(opts)
```
Comment (tworec) [inline nbdime/gitdiffdriver.py]: much better

Comment (vidartf) [inline nbdime/gitdiffdriver.py]: The solution that was there was correct given the API, but not very pretty. So I refactored the API to something a little better :)

File: nbdime/args.py (hunk 1 above)
Comment (vidartf) [inline nbdime/args.py]: (had to push a fix for it)

File: nbdime/gitdiffdriver.py (hunk 2 above)
Comment (tworec) [inline nbdime/gitdiffdriver.py]: ;)

**Comment by tworec** [comment]
Do we need anything more to merge this PR?

**Comment by vidartf** [comment]
I would have preferred some extra eyes on it if possible, but I'll merge for now, and make any changes in a separate PR later if needed.

**Pull request merged**